workshops/models.py

```diff
@@ -397,6 +400,42 @@ def is_staff(self):
     def is_admin(self):
         return is_admin(self)
 
+    @property
+    def missing_swc_instructor_requirements(self):
+        """ Returns set of requirements that are not passed yet by the
+        trainee and are mandatory to become SWC Instructor.. """
+        training = TrainingRequirement.objects.get(name='Training')
+        homework = TrainingRequirement.objects.get(name='SWC Homework')
+        discussion = TrainingRequirement.objects.get(name='Discussion')
+        demo = TrainingRequirement.objects.get(name='SWC Demo')
+
+        requirements = {training, homework, discussion, demo}
+        progresses = self.trainingprogress_set.filter(discarded=False, state='p')
+        passed_requirements = {p.requirement for p in progresses}
+        return requirements - passed_requirements
+
+    @property
+    def missing_dc_instructor_requirements(self):
+        """ Returns set of requirements that are not passed yet by the
+        trainee and are mandatory to become DC Instructor.. """
```

Comment:
BTW, PEP257 doesn't say anything about putting space after starting `"""` and before ending `"""`.